src/Hl7.Fhir.Base/Specification/Terminology/MimeTypeTerminologyService.cs

```diff
@@ -0,0 +1,156 @@
+            }
+            catch (Exception e)
+            {
+                //500 internal server error
+                throw new FhirOperationException(e.Message, (HttpStatusCode)500);
+            }
+
+        }
+
+        private async Task<Parameters> validateCodeVS(Coding coding)
+        {
+            return await validateCodeVS(coding.Code, coding.System).ConfigureAwait(false);
+        }
+
+        private async Task<Parameters> validateCodeVS(CodeableConcept cc)
+        {
+            var result = new Parameters();
+
+            // Maybe just a text, but if there are no codings, that's a positive result
+            if (!cc.Coding.Any())
+            {
+                result.Add("result", new FhirBoolean(true));
+                return result;
+            }
+
+            // If we have just 1 coding, we better handle this using the simpler version of ValidateBinding
+            if (cc.Coding.Count == 1)
+                return await validateCodeVS(cc.Coding.Single()).ConfigureAwait(false);
+
+
+            // Else, look for one succesful match in any of the codes in the CodeableConcept
+            var callResults = await Task.WhenAll(cc.Coding.Select(coding => validateCodeVS(coding))).ConfigureAwait(false);
+            var anySuccesful = callResults.Any(p => p.GetSingleValue<FhirBoolean>("result")?.Value == true);
+
+            if (anySuccesful == false)
+            {
+                var messages = new StringBuilder();
+                messages.AppendLine("None of the Codings in the CodeableConcept were valid for the binding. Details follow.");
+
+                // gathering the messages of all calls
+                foreach (var msg in callResults.Select(cr => cr.GetSingleValue<FhirString>("message")?.Value).Where(m => m is { }))
+                {
+                    messages.AppendLine(msg);
+                }
+
+                result.Add("message", new FhirString(messages.ToString()));
+                result.Add("result", new FhirBoolean(false));
+            }
+            else
+            {
+                result.Add("result", new FhirBoolean(true));
+            }
+
+            return result;
+        }
+
+        private static Task<Parameters> validateCodeVS(string? code, string? system)
+        {
+            var result = new Parameters();
+
+            if (code is null)
+            {
+                result.Add("message", new FhirString("No code supplied."))
+                      .Add("result", new FhirBoolean(false));
+
+            }
+            else if (system == MIMETYPE_SYSTEM || system == null)
+            {
+                var success = validateMimeType(code);
+
+                if (success)
+                {
+                    result.Add("result", new FhirBoolean(true));
+                }
+                else
+                {
+                    result.Add("result", new FhirBoolean(false))
+                          .Add("message", new FhirString("Invalid mimetype specified"));
```

Comment:
I'd change this to `XXXXX is not a valid mimetype.`  Always end errors with periods, it's a message for humans.